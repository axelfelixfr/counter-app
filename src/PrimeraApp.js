// import React, { Fragment } from 'react'; // Importamos Fragment para retornar diversos elementos
import React from 'react';
import PropTypes from 'prop-types'; // Importamos las properties

// Hoy en día se trabaja con Functional Component, anteriormente la forma de trabajar era con Class Component
const PrimeraApp = ({saludo = 'Hola Mundo!!!', nuevaProp, subtitulo}) => {
    // Podemos aplicar desestructuración de las props con las {}
    // Si el componentes no llega a tener ese elemento en las props, podemos asignarle uno por defecto como saludo = "Hola Mundo!!!"

    // NO se pueden imprimir en HTML variables con valores booleanos, como por ejemplo, "valor = true"
    // const saludo = 'Hola mundo';
    const objeto = {
        nombre: 'Axel Félix',
        edad: 21
    }

    return (
        /* Para no importar el componente Fragment, se puede utilizar solo <></> */
        <>
            {/* La constante saludo ahora vienen de las props */}
            <h1>{ saludo }</h1>
            <h2>{ nuevaProp }</h2>
            {/* El método stringify es para convertir el objeto en un string */}
            <pre>{ JSON.stringify(objeto, null, 3) }</pre>
            <p>{ subtitulo }</p>
        </>
    );

};

// Para darle tipado a las properties usar el objeto propTypes de la librería que importamos
PrimeraApp.propTypes = {
    // Podemos especificar el tipado de las properties
    saludo: PropTypes.string,
    // Podemos especificar si es obligatoria la prop con isRequired
    nuevaProp: PropTypes.number.isRequired
}

// Para definir props por defecto puede ser arriba al definir el componente
// O también usar el objeto defaultProps
PrimeraApp.defaultProps = {
    // A la prop subtitulo le damos el valor por defecto de 'Mi primera aplicación'
    subtitulo: 'Mi primera aplicación'
}

export default PrimeraApp;