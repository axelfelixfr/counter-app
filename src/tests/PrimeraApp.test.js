import React from 'react';
// import { render } from '@testing-library/react'; // Comentamos el render ya que no ocupamos el testing-library
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import PrimeraApp from '../PrimeraApp';

// De esta forma podemos hacer las pruebas con la librería que viene por defecto con create-react-app
describe('Pruebas en <PrimeraApp />', () => {
    
    /* Comentamos esta prueba ya que no tenemos importado la librería de testing-library que contiene el método toBeInTheDocument, usado en la prueba
    test('Debe de tener la nuevaProp y ser de tipo number', () => {
        const nuevaProp = 123;
        const { getByText } = render(<PrimeraApp nuevaProp={nuevaProp} />);
        // Se recomienda no usar los métodos por defecto de la librería testing-library de react
        expect(getByText(nuevaProp)).toBeInTheDocument();
        // Verificamos que en PrimeraApp este la nuevaProp con toBeInTheDocument()
    })
    */

    test('Debe de mostrar <PrimeraApp /> correctamente', () => {
        const nuevaProp = 123;
        const wrapper = shallow(<PrimeraApp nuevaProp={nuevaProp} />);

        expect(wrapper).toMatchSnapshot();
        // Comparamos el componente PrimeraApp con el snapshot
    });

    test('Debe de mostrar el subtitulo enviado por props', () => {
        const nuevaProp = 123;
        const subTitulo = 'Mi primera aplicación';
        const wrapper = shallow(<PrimeraApp nuevaProp={nuevaProp} subtitulo={subTitulo} />);
        // Con el objeto wrapper podemos hacer un querySelector() como en JavaScript con el método find()
        const textoParrafo = wrapper.find('p').text();
        // Buscamos el parrafo y extraemos el texto que contiene

        expect(textoParrafo).toBe(subTitulo); // Con expect y toBe comparamos que el contenido del parrafo sea igual a subTitulo
        // Esperamos(textoParrafo).seaIgual(subTitulo)
    });
});