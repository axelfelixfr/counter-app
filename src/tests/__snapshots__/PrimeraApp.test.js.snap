// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> Debe de mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola Mundo!!!
  </h1>
  <h2>
    123
  </h2>
  <pre>
    {
   "nombre": "Axel Félix",
   "edad": 21
}
  </pre>
  <p>
    Mi primera aplicación
  </p>
</Fragment>
`;
